apps/site/pages/en/learn/typescript/publishing-a-ts-project.md

```diff
@@ -0,0 +1,218 @@
+---
+title: Publishing a TypeScript project
+layout: learn
+authors: JakobJingleheimer
+---
+
+# Publishing a TypeScript project
+
+This article augments TypeScript's own [Publishing guide](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) with specifics for native node support.
+
+Some important things to note:
+
+- Everything from [Publishing a package](../modules/publishing-a-package) applies here.
+
+- Node runs TypeScript code via a process called "[type stripping](https://nodejs.org/api/typescript.html#type-stripping)", wherein node (via [Amaro](https://github.com/nodejs/amaro)) removes TypeScript-specific syntax, leaving behind vanilla JavaScript (which node already understands). This behaviour is enabled by default as of node version 23.6.0.
+
+  - Node does **not** strip types in `node_modules` because it can cause significant performance issues for the official TypeScript compiler (`tsc`) and parts of VS Code, so the TypeScript maintainers would like to discourage people publishing raw TypeScript, at least for now.
```

Comment:
```suggestion
- Node does **not** strip types in `node_modules` because it can cause significant performance issues for the official TypeScript compiler (`tsc`) and parts of VS Code, so the TypeScript maintainers would like to discourage people publishing raw TypeScript, at least for now.
```

Reply:
The indentation wasn't an accident/typo, if that's what you're thinking. This could be root-level I suppose, but I was thinking it was an addendum to the item above it. I'm splitting hairs though don't feel strongly. If people thing it's better to keep root-level, sure.

Reply:
I mean more like using a connecting lines like so

```
a/
├── b/
│   └── c
└── d
```

whereas you have something closer to

```
a/
├── b/
    └── c
└── d
```

Reply:
Yes, I understood 😉 That is what I originally had, and it renders wonky on nodejs.org.

Reply:
Sorry, I misread and responded to the wrong comment. 😅